File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> people;
    
    for(string part : participant){
        people[part]++;
    }
    
    for(int i=0; i<completion.size();i++){
        if(people[completion[i]]>0){
            people[completion[i]]--;
        }
    }
    
    for(auto p: people){
        if(p.second!=0){
            answer = p.first;
            break;
        }
    }
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.io.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        // participant에는 있는데 completion에는 없는 사람
        HashMap<String, Integer> map = new HashMap<>();
        for(int i=0; i<participant.length; i++){
            map.put(participant[i], map.getOrDefault(participant[i], 0)+1);
        }
   
        
        for(String comple : completion){
            // completion 돌면서 hashmap에 있는지 확인
            if(map.containsKey(comple)){
                map.put(comple, map.get(comple)-1);
            }
        }
        
        for(String key: map.keySet()){
            if(map.get(key) != 0){
                answer = key;
            }
        }
        
    return answer;
    }
}